src/normalizations/cms/useNormalizedCMSResults.js

```diff
@@ -20,11 +20,73 @@ export const EDITORIAL_FIELD_TYPE_VALUES = {
 }
 
 // Drupal JSONapi encodes `&` in URLs, which React can't handle https://github.com/facebook/react/issues/6873#issuecomment-227906893
-function cleanupDrupalUri(uri) {
-  return uri.replace(/&amp;/g, '&')
+const cleanupDrupalUri = (uri) => uri.replace(/&amp;/g, '&')
+
+export const getLocaleFormattedDate = ({
+  field_publication_date,
+  field_publication_day,
+  field_publication_year,
+  field_publication_month,
+} = {}) => {
+  const year = parseInt(field_publication_year, 10)
+  const yearIsValidNumber = Number.isNaN(year) === false
+  const hasPossibleValidDate = field_publication_date?.length > 0 || yearIsValidNumber
+
+  if (!hasPossibleValidDate) {
+    return {
+      localeDate: '',
+      localeDateFormatted: '',
+    }
+  }
+
+  if (field_publication_date) {
+    return {
+      localeDate: field_publication_date,
+      localeDateFormatted: formatDate(new Date(field_publication_date.replace(' ', 'T'))),
+    }
+  }
+
+  const day = parseInt(field_publication_day, 10)
+  const monthIndex = parseInt(field_publication_month, 10)
+
+  const dateParts = [
+    year,
+    !Number.isNaN(monthIndex) && monthIndex - 1,
+    !Number.isNaN(day) && day,
+  ].filter(Boolean)
```

Comment:
Since months are zero indexed in JavaScript dates would this not lead to a missing month number for January? If `monthIndex` is 1 (so January) and `monthIndex - 1` (so `0`) is passed to the `Boolean` filter will this not result in it getting filtered out?

In this case I think a procedural approach here might be more clear and less error prone:

```suggestion
  const dateParts = [year]
  
  if (!Number.isNaN(monthIndex)) {
    dateParts.push(monthIndex - 1)
  }
  
  // Also check if the month was pushed before, we don't
  // want to accidentally add a day in the month position.
  if (dateParts.length === 2 && !Number.isNaN(day)) {
    dateParts.push(day)
  }
```

Reply:
Good one; I'll add a test case for that.

Reply:
Solved it in a different way; a bit cleaner than adding conditions imho

Reply:
Nice, long as we cover this edge case I am happy :smile: